### PR TITLE
[backend/frontend] enforce null consistency with executor and command (#1621)

### LINKED ISSUE #1621: Payload - Enforce required cleanup command AND executor when choosing one of the two
## Description

When I create a payload, I can specify a cleanup executor without a cleanup command.
And vice versa.

## Expected Output

If I specify a cleanup executor or a cleanup command, the second property should be defined too.

### PR DESCRIPTION
**Pull request by antoinemzs**
### Proposed changes

* Enforce having a value for both the Cleanup Executor and the Cleanup Command in Payload definition

### Related issues

* Closes #1621
* Must be merged at the same time: https://github.com/OpenBAS-Platform/collectors/pull/49

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

On the payload definition panel, it is now not possible to only define the cleanup executor or the cleanup command. Both must be defined at the same time or none of them (i.e. blank fields).

The backend part was done with a "pure" null philosophy, unfortunately the frontend is not yet able to handle null values in react field components: the frontend special cases empty string for fields to submit null instead.

### CODE REVIEW

File: openbas-front/src/utils/Localization.js [hunk 1]
```diff
@@ -2680,6 +2681,7 @@ const i18n = {
       'Update the security platform': '更新安全平台',
       'Security Platforms': '安全平台',
       'Content should not be empty': '内容不可为空',
+      'Cleanup command and executor must be defined together or none at all': '清理命令和执行器必须一起定义，否则根本无法定义',
```
Comment (antoinemzs) [inline openbas-front/src/utils/Localization.js:2684]: Deepl translation

**Comment by antoinemzs** [comment]
Issue found: the new constraint prevents importing the atomic red team database

**Comment by antoinemzs** [comment]
> Issue found: the new constraint prevents importing the atomic red team database

fixed

**Comment by savacano28** [comment]
**nitpick:** When I try to update a payload (image 1), I receive errors for all the mandatory fields, but not for the scenario where only one of two fields is filled, so maybe it is possible to add this verification?

image

File: openbas-api/src/test/java/io/openbas/rest/PayloadApiTest.java [hunk 2]
```diff
@@ -158,4 +279,20 @@ private PayloadCreateInput getExecutablePayloadCreateInput() {
     input.setExecutableArch(Endpoint.PLATFORM_ARCH.x86_64);
     return input;
   }
+
+  private PayloadCreateInput getCommandLinePayloadCreateInput() {
+    PayloadCreateInput input = new PayloadCreateInput();
```
Comment (savacano28) [inline openbas-api/src/test/java/io/openbas/rest/PayloadApiTest.java:284]: **nitpick**: Maybe use a fixture?

File: openbas-api/src/main/java/io/openbas/migration/V3_47__Enforce_payload_command_definition_consistency.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  @Override
+  public void migrate(Context context) throws SQLException {
+    Statement migrator = context.getConnection().createStatement();
+    migrator.execute(
+        """
+                    UPDATE payloads
+                    SET command_executor = NULL, command_content = NULL
+                    WHERE (command_executor = '') IS NOT FALSE OR (command_content = '') IS NOT FALSE
+                    """);
+    migrator.execute(
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/migration/V3_47__Enforce_payload_command_definition_consistency.java:21]: **praise**: Thanks for adding this constraint

File: openbas-api/src/test/java/io/openbas/rest/PayloadApiTest.java (hunk 2 above)
Comment (antoinemzs) [reply openbas-api/src/test/java/io/openbas/rest/PayloadApiTest.java:284]: Ah, I see that this is the *Input DTO-type object, not the actual model. This is probably why I haven't used the fixture in the first place, which returns a Payload model (not a PayloadInput DTO). I'll see if I can "fixturise" this.

**Comment by antoinemzs** [comment]
> **Question**: When I create a payload with a type other than 'executable type,' I receive all the expected errors, but for 'executable type,' I don't. Could you please verify why the behavior is different?

It seems that if the `Executable` field is in error, the validation is not run on the remainder of the form fields, e.g. the Cleanup command executor/command tuple.

**Comment by antoinemzs** [comment]
> **nitpick:** When I try to update a payload (image 1), I receive errors for all the mandatory fields, but not for the scenario where only one of two fields is filled, so maybe it is possible to add this verification?

@savacano28 could not reproduce this behaviour locally, unsure whether it's still occurring?

**Review by savacano28 — APPROVED** [review]

**Pull request merged**